samples/observability-traces.js

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Copyright 2024 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// sample-metadata:
+//   title: Observability (Tracing) with OpenTelemetry
+//   usage: node observability-traces.js <PROJECT-ID> <INSTANCE-ID> <DATABASE-ID>
+
+'use strict';
+
+// Setup OpenTelemetry and the trace exporter.
+// [START spanner_trace_and_export_spans]
+const {Resource} = require('@opentelemetry/resources');
+const {
+  NodeTracerProvider,
+  TraceIdRatioBasedSampler,
+  // eslint-disable-next-line n/no-extraneous-require
+} = require('@opentelemetry/sdk-trace-node');
+// eslint-disable-next-line n/no-extraneous-require
+const {BatchSpanProcessor} = require('@opentelemetry/sdk-trace-base');
+const {
+  SEMRESATTRS_SERVICE_NAME,
+  SEMRESATTRS_SERVICE_VERSION,
+  // eslint-disable-next-line n/no-extraneous-require
+} = require('@opentelemetry/semantic-conventions');
+
+const resource = Resource.default().merge(
+  new Resource({
+    [SEMRESATTRS_SERVICE_NAME]: 'spanner-sample',
+    [SEMRESATTRS_SERVICE_VERSION]: 'v1.0.0', // The version of your app running.,
+  })
+);
+
+/*
+ * Uncomment these lines to debug OpenTelemetry.
+const {diag, DiagConsoleLogger, DiagLogLevel} = require('@opentelemetry/api');
+diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.ALL);
+*/
+
+// Create the Google Cloud Trace exporter for OpenTelemetry.
+const {
+  TraceExporter,
+} = require('@google-cloud/opentelemetry-cloud-trace-exporter');
+const exporter = new TraceExporter();
+
+// Optionally, you can enable gRPC instrumentation, by removing this guard.
+if (process.env.SPANNER_ENABLE_GRPC_INSTRUMENTATION === 'true') {
+  // eslint-disable-next-line n/no-extraneous-require
+  const {registerInstrumentations} = require('@opentelemetry/instrumentation');
+  const {GrpcInstrumentation} = require('@opentelemetry/instrumentation-grpc');
+  registerInstrumentations({
```

Comment:
How does this work `registerInstrumentations` . Does it need trace provider to be registered globally. Will this work when we inject local trace provider ?

Reply:
I showed in the sample how to pass in the tracerProvider.